### ISSUE TITLE
What are the differences between the packages ltsp-server and ltsp-server-standalone 

### ISSUE THREAD

**Issue by dashohoxha** [docker-scripts/dev--LTSP#51]
How can we find the differences between two similar Ubuntu packages, or between a Debian package and an Ubuntu package that are similar?

**Comment by d78ui98** [comment]
LTSP-server - It package just provides the basic  minimal features necessary to build an ltsp environment.
depends - `debconf, debootstrap, openssh-client` 
recommends - `nbd-server, dnsmasq, openssh-server, squashfs-tools`

LTSP-server-standalone - It is complete ltsp server that includes ltspfs, ltsp-server package, dnsmasq.
depends - `dnsmasq, ldm-server, ltsp-server, ltspfs, nbd-server, openssh-server, squashfs-tools`
recommends - `ltsp-docs`

I downloaded their .deb packages with `apt-get` and used `debdiff` to compare them
here is a gist of the output - https://gist.github.com/d78ui98/8159947236b0245e5145fd8f90bb48ce

I also extracted the source of the their packages. I am using `meld` to see differences between them
here is the screenshot - https://image.ibb.co/kDkBKo/comparision.png

**Comment by dashohoxha** [comment]
Try to use `diff -u` on the source code of these packages. Neither the gist, nor the screenshot do not seem very useful to me.

Then, what are the differences between these packages? Should we use one of them or the other?
Have you tried to use each of them? How are they different from each-other?
The best way to find out is by testing them, but you can also ask on the mailing list, in case somebody knows.

**Comment by d78ui98** [comment]
I used `diff -u`. Got lots of output that concludes there are lots of bash scripts inside ltsp-server source. We can take a look at them if we wanter deeper understanding of ltsp.

> what are the differences between these packages?

well `ltsp-server` is the main packages. Most stuff like man pages, ltps-build-client files for different distributions, all important scripts, etc are defined in it.  

The ltsp-server-standalone package does not have a lot of files. It has defined ltsp-server as one of its dependencies in `control` file. That way everything inside ltsp-server packages used in standalone. Standalone also has a lot of other dependencies like `gnome-session, mate-desktop-environment. x-session-manager, x-window-manager, isc-dhcp-server, dhcp3-server, dnsmasq, ldm-server, libasound2-plugins, ltspfs, ltsp-server, nbd-server`. 

>  Should we use one of them or the other?

I personally think we should always use `ltsp-server` instead of `ltsp-server-standalone`. Since it is very flexible as after installing it I can install the only packages that we need. This also makes it resouce friendly. 

Standalone is heavy. It also confuses me a bit. I mean why it installs 3 packages( `isc-dhcp-server, dhcp3-server, dnsmasq`) that does exact same stuff. isc-dhcp-server is not even used anymore. Still one can go for standalone packages since it requires less number of steps. 

> Have you tried to use each of them?

 yes. First ever ltsp setup that I created was with `ltsp-server-standalone`.  I remember it had very less number of steps.

- just have to install ltsp-server-standalone package.
- it creates `/etc/ltsp` directory inside it edit `dhcpd.conf`
- then run ltsp-build-client
- after it reboot and start the client

and with ltsp-server package we have been creating it ever since.

>  but you can also ask on the mailing list, in case somebody knows.

Yes I can drop a mail there. If someone would like to add to this.
